src/lib/Embed.hs

```diff
@@ -601,7 +635,24 @@ traverseAtom def@(_, _, fAtom) atom = case atom of
     items' <- traverse fAtom items
     return $ VariantTy $ Ext items' rest
   ACase e alts ty -> ACase <$> fAtom e <*> mapM traverseAAlt alts <*> fAtom ty
+  DataConRef dataDef params args -> DataConRef dataDef <$>
+    traverse fAtom params <*> traverseNestedArgs args
+  BoxedRef b ptr size body -> do
+    ptr'  <- fAtom ptr
+    size' <- buildScoped $ evalBlockE def size
+    -- Is this what we want? We can't recur because we don't want decls
+    Abs b' body' <- substEmbedR $ Abs b body
```

Comment:
The `Abs` we want here is an `Abs Binder Atom`, not an `Abs Binder Block`. That might need to change.

Reply:
I think we should just `buildAbs` followed by a partial cast of a block back to an atom. We already do this when we traverse an `ACase`. It's an error for a traversal to emit decls inside those, but it should still have a possibility to see what happens inside them. Otherwise the traversals we use to e.g. count uses of a variable would be broken!